express-get-json/server.ts

```diff
@@ -0,0 +1,48 @@
+import express from "express";
+
+
+interface Person {
+  id: number;
+  name: string;
+  course: string;
+  score: number;
+}
+
+interface grades {
+  12: Person,
+  47: Person,
+  273: Person
+}
+
+const grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95,
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100,
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92,
+  },
+};
+
+const gradesArray: object[] = [];
+for (const grade in grades) {
+  gradesArray.push(grades[grade]);
+}
+
+const app = express();
+
+app.listen(8080, () => console.log('Listening on Port 8080'))
+app.get('/api/grades', (req, res) => {
+    res.json(gradesArray);
```

Comment:
You should do all the work you did above inside this function. With Express endpoints, you almost always want to 
- read any data you need
- perform any operations
- write the results
- send the response

Keep in mind that an Express server stays up and runs continuously, responding to many, many requests. But the code at the top level (such as lines 38-41) executes only the first time the server is loaded. So it's safest to always read the values before you work on them.